File: pfib.py
from multiprocessing import Process

#@micropython.viper
#def fib(n: int) -> int:
def fib(n):
  if n < 3:
    return 1
  return int(fib(n-2)) + int(fib(n-1))

def fibTest(n):
  print(n, ":", fib(n))

procs = [Process(target=fibTest, args=(n,))
         for n in range(34, 44)]
for p in procs:
  p.start()
for p in procs:
  p.join()

File: pfib.rs
//rustc -O -C target-cpu=broadwell pfib.rs

use std::thread;

fn fib(n: u64) -> u64
{
  if n < 3 { 1 } else { fib(n-1) + fib(n-2) }
}

fn fib_test(n: u64)
{
  println!("fib({}) = {}", n, fib(n));
}

fn main()
{
  let mut thrds = Vec::new();
  for x in 44 .. 49
  {
    thrds.push(
      thread::spawn(move || fib_test(x) ));
  }
  for t in thrds
  {
    t.join().unwrap();
  }
}

